tests/test_buildoptionsparser.py

```diff
@@ -1,9 +1,13 @@
 import pytest
-from iotedgedev.buildoptionsparser import BuildOptionsParser
 
-pytestmark = pytest.mark.unit
+from .version import test_py2, minversion
+
+if not test_py2:
```

Comment:
Many libraries are py3 specific. This prevents tests from failing and allows me to raise an expected assertion error on the version catch in iotedgedev/version_catch.py

Reply:
I don't quite understand this comment, can you please expand on what is failing? 

By updating the `python_requires` constraint pip should only install the package in the supported range and error out otherwise. Then you can apply version or env constraints to package dependencies or requirements file entries if needed.

It would be good to use as much of the available existing tooling as possible.

Reply:
The idea was that even if pip install fails, tox and pytest will be installed and can still run. This solution provides a catch for testing, avoiding all tests except for the e2e test I built to get to the version catch

Reply:
OK I see. Testing infrastructure wise (like via tox) you should be able to configure tox to properly install the package in the defined compatible python runtimes and abort the run if a failure occurs in package installation. Then philosophically, this package defines its python run requirements via attributes like `classifiers` and `python_requires`, which is a standard approach for the Python ecosystem and the Python ecosystem tools + users should abide. In general, if the package claims its 3.6+ and a user tries to run aspects in a non-compatible environment then failure is expected.

That being said, to still achieve the goal here, but in less code consider refactoring to use one of the following pytest configuration approaches:
* [collect_ignore_glob or collect_ignore](https://docs.pytest.org/en/latest/reference.html#globalvar-collect_ignore_glob)
* A wider scoped fixture...something like

```python
@pytest.fixture(scope="session", autouse=True)
def ensure_python_version():
    if sys.version_info < (3, 6):
        pytest.skip("Incompatible environment. The package requires Python 3.6+.")
```

Reply:
This is exactly the kind of thing I was looking for! I wanted a global way to skip tests but could not find one. 

I will investigate configuring tox properly first and will use this as backup.

Thanks @digimaun!